accord-core/src/main/java/accord/messages/Commit.java

```diff
@@ -173,64 +174,65 @@ public static void stableAndRead(Node node, Topologies all, Kind kind, TxnId txn
     {
         Invariants.checkState(all.oldestEpoch() == txnId.epoch());
         Invariants.checkState(all.currentEpoch() == executeAt.epoch());
-        Topology executes = all.forEpoch(executeAt.epoch());
-        Topology coordinates = all.forEpoch(txnId.epoch());
+        Topology executes = all.getEpoch(executeAt.epoch());
+        Topology coordinates = all.getEpoch(txnId.epoch());
 
         SortedArrayList<Id> contact = all.nodes().without(all::isFaulty);
-        sendTo(contact, readSet, (set, id) -> set.contains(id.id), readScope, node, coordinates, executes, all, kind, Ballot.ZERO,
+        sendTo(contact, readSet, (set, id) -> set.contains(id.id), (set, id) -> false, readScope, node, coordinates, executes, all, kind, Ballot.ZERO,
```

Comment:
Did you intend to add this _after_ `(set, id) -> set.contains(id.id)`?

Reply:
No, but perhaps it is counter-intuitive. If isRead is true, then we always register a callback (as anything else would make no sense). So we only invoke this for non-reads.